File: src/main/java/se/lars/ApiUser.java
package se.lars;

import io.vertx.core.AsyncResult;
import io.vertx.core.Handler;
import io.vertx.core.json.JsonObject;
import io.vertx.ext.auth.AbstractUser;
import io.vertx.ext.auth.AuthProvider;


public class ApiUser extends AbstractUser
{
    private JsonObject _principal;

    public ApiUser(JsonObject principal)
    {
        _principal = principal;
    }

    @Override
    protected void doIsPermitted(String permission, Handler<AsyncResult<Boolean>> resultHandler)
    {

    }

    @Override
    public JsonObject principal()
    {
        return _principal;
    }

    @Override
    public void setAuthProvider(AuthProvider authProvider)
    {

    }
}

File: src/main/kotlin/se/lars/auth/ApiUser.kt
package se.lars.auth

import io.vertx.core.AsyncResult
import io.vertx.core.Handler
import io.vertx.core.json.JsonObject
import io.vertx.ext.auth.AbstractUser
import io.vertx.ext.auth.AuthProvider


class ApiUser(private val _principal: JsonObject) : AbstractUser() {


    override fun doIsPermitted(permission: String, resultHandler: Handler<AsyncResult<Boolean>>) = Unit

    override fun principal() = _principal

    override fun setAuthProvider(authProvider: AuthProvider) = Unit
}
